Simple Incremental/Assets/Scripts/Monobehaviours/ObjectPooler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+/* This class is used for pooling objects and reusing objects for performance
+ * This works best when lots of the same object are spawned multiple times
+ * such as projectiles from a gun */
+public class ObjectPooler : MonoBehaviour
+{
+    public static ObjectPooler instance;
+    public Dictionary<string, Queue<GameObject>> dict = null;
+
+    void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+            dict = new Dictionary<string, Queue<GameObject>>();
+        }
+        else
+        {
+            Destroy(this);
+        }
+    }
+
```

Comment:
You previously had this code:

```
void Start()
    {
        foreach (GameObject obj in objectsToPool)
        {
            AddPoolObjects(obj, initialPoolSize);
        }
    }
```

and

```
    private void AddPoolObjects (GameObject go, int amountToAdd)
    {
        string name = go.name;
        if (!dict.ContainsKey(name))
        {
            dict.Add(name, new Queue<GameObject>());

        }
        for (int i = 0; i < amountToAdd; i++)
        {
            GameObject obj = (GameObject)Instantiate(go);
            obj.name = name;
            obj.transform.SetParent(transform);
            obj.SetActive(false);
            dict[name].Enqueue(obj);
        }
    }
```

There was nothing wrong with initializing the objects. However, since AddPoolObjects was only used in start, extracting it into a separate method might not be necessary.

Additionally, you're setting the object name, the parent, and the object activity in your AddPoolObjects. These responsibilities should really be placed on the ObjectPool client. (Your prior logic was basic on all object's having the same name, so it was needed, but now it's not :). New logic is based on the prefab name)

I've added a new implementation for prepooling objects that involves a struct

Reply:
My code was calling it from a separate location as well. I wanted the idea that if you hit a min number of objects you could pool a batch of objects. It seemed that might be useful in case there was a case where a huge number of objects were spawned. Instead of initializing each one as they spawn, initializing a tunable block based on performance. I'm not sure how it would affect gameplay but I figured it might be useful. Regardless its not a big deal.